cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -120,24 +122,113 @@ def behavior_for_path(behaviors, path)
 
 # Generates the logic string for the specified behavior.
 def process_behavior(behavior, app, section)
-  if section == 'proxy'
-    process_proxy(behavior[:proxy] || app, app)
-  else
-    process_cookies(behavior[:cookies], section)
+  case section
+    when 'proxy'
+      process_proxy(behavior[:proxy] || app, app)
+    when 'vary'
+      process_vary(behavior)
+    when 'request'
+      process_request(behavior)
+    else
+      process_response(behavior)
   end
 end
 
-# Returns the cookie-filter string for a given 'cookies' behavior.
-def process_cookies(cookies, section)
-  if section == 'request'
-    return '# Allow all request cookies.' if cookies == 'all'
-    cookies = ['NO_CACHE'] if cookies == 'none'
-    "cookie.filter_except(\"#{cookies.join(',')}\");"
-  else
-    cookies == 'none' ?
-      'unset beresp.http.set-cookie;' :
-      '# Allow set-cookie responses.'
+# VCL string to create or update a Vary header field with the provided Vary header.
+def set_vary(header, resp)
+  <<VCL
+  if (!#{resp}.http.Vary) {
+    set #{resp}.http.Vary = "#{header}";
+  } elseif (#{resp}.http.Vary !~ "(?<=\\s|,|^)#{header}") {
```

Comment:
This will fail to add a header if the new header is a prefix of an already added one.  Suppose there was a header called FooBar and we added one called Foo; we wouldn't both to add it.

I think some appending you expression  like (\s*(,|(\s*$))) would do the trick.  

Or if I was using Ruby I would split on /\s*,\s*/ and make sure the array doesn't contain the header.